pkg/reftracker/app_update_status.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package reftracker
+
+import (
+	"fmt"
+	"sync"
+)
+
+type AppUpdateStatus struct {
+	lock               sync.Mutex
+	appsToUpdateStatus map[string]struct{}
+}
+
+func NewAppUpdateStatus() *AppUpdateStatus {
+	return &AppUpdateStatus{appsToUpdateStatus: map[string]struct{}{}}
+}
+
+func (a AppUpdateStatus) MarkNeedsUpdate(appName, namespace string) {
```

Comment:
any reason not to use refkeys here as well?

Reply:
Changed to RefKey